virtio-win drivers sign check Test

peixiu opened the pull request:
1. Boot a guest with latest virtio-win.iso and virtio-win.vfd files
2. Run signtool.exe verify for each driver '.sys', '.inf' and 'Wdf**.dll'files.

id:1375853

Signed-off-by: peixiu <phou@redhat.com>

peixiu commented:
Tested with win2008, win7, win8, win10,win2012,win2016 guests, it can pass on all guests. 
Results as follows:

 (1/4) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.i386.io-github-autotest-qemu.virtio_driver_sign_check: PASS (447.86 s)
 (2/4) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2008.i386.sp2.io-github-autotest-qemu.virtio_driver_sign_check: PASS (773.09 s)
 (3/4) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.i386.io-github-autotest-qemu.virtio_driver_sign_check: PASS (119.33 s)
 (4/4) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win8.i386.1.io-github-autotest-qemu.virtio_driver_sign_check:PASS (101.69 s)

PASS 1-smp_2.8192m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win7.x86_64.sp1.io-github-autotest-qemu.virtio_driver_sign_check
PASS 11-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2008.i386.sp2.io-github-autotest-qemu.virtio_driver_sign_check/
PASS 10-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.virtio_driver_sign_check/
PASS 12-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2008.x86_64.sp2.io-github-autotest-qemu.virtio_driver_sign_check/
PASS 13-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2008.x86_64.r2.io-github-autotest-qemu.virtio_driver_sign_check/
PASS 14-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2012.x86_64.io-github-autotest-qemu.virtio_driver_sign_check/
PASS 15-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2012.x86_64.r2.io-github-autotest-qemu.virtio_driver_sign_check/
PASS 16-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win2016.x86_64.io-github-autotest-qemu.virtio_driver_sign_check/

lijinlijin commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,88 @@
-    cdrom_virtio = isos/windows/virtio-win.iso
-    cdrom_sdk = isos/windows/GRMSDKX_EN_DVD.iso
+    cdroms += " virtio signtool"
+    cdrom_virtio = isos/windows/virtio-win-latest-signed-el7.iso
```
we also run this case on prewhql(unsigned) build,not only signed package

peixiu replied:
ok, will change it, thanks~

lijinlijin commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,88 @@
+            signtool = "WIN_UTILS:\signtool\signtool-HLK\Signtool_amd64_HLK.exe"
+            arch_dir = 64
+        os_dirs = Win10 2k16
+    signtool_cmd = ${signtool} verify /v /kp /c %s %s
```
floppy_name and signtool parameters of each widows platforms are kind of duplicated,please re-arrange them.

lijinlijin replied:
so is arch_dir

peixiu replied:
ok, will re-arrange it, thanks~

peixiu commented:
Tested with latest code:
PASS 1-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.i386.io-github-autotest-qemu.virtio_driver_sign_check
PASS 2-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Win10.x86_64.io-github-autotest-qemu.virtio_driver_sign_check

peixiu commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,18 +1,20 @@
+       wdf***.dll file, like balloon/viorng/vioserial/pvpanic.
+       Second type driver with wdf***.dll and 2 .sys files, only vioinput;
+       Third type drivers without wdf***.dll file,like netkvm/vioscsi/viostor.
     3) use SignTool.exe to verify whether block driver digital signed
```
has change it, thanks~

kimi1978wy commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,121 @@
+
+        cat_sys = map(lambda x: (cat_element, x), driver_sys)
+        cat_inf = map(lambda x: (cat_element, x), driver_inf)
+        cat_wdf = map(lambda x: (cat_element, x), driver_wdf)
```
maybe these codes above can be written in a loop

peixiu replied:
Hi wyu, 
Here, it is not suitable use a loop.  Because it doesn't need to judge conditions, so while loop  is not suitable;  and if use "for loop", the loop format like "for tmp in list/string:", the tmp should be a string or character, but in our code, the drivers['sys']  is a list, and the drivers is a dict, they are not suitable to use "for loop" .

kimi1978wy commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,121 @@
-        else:
-            return False
-
     timeout = float(params.get("login_timeout", 240))
```
maybe login_timeout=360 is better.

peixiu replied:
that's ok, thanks~

xiagao commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,121 @@
+                status, result = session.cmd_status_output(test_cmd)
+                if status:
+                    msg = "Fail command: %s. Output: %s" % (test_cmd, result)
                     raise error.TestFail(msg)
```
peixiu, pls change to test.fail() Incidentally

peixiu replied:
has updated, thanks~

xiagao commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,121 @@
+    need_test_driver = params.objects("tested_drivers")
+    get_disk_letter = params.get("get_disk_letter_cmd")
+
+    def cat_filter(driver):
```
suggest to add docstring to make the function clearly

peixiu replied:
has updated, thanks~

xiagao commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,121 @@
+                if driver_dir in need_test_driver:
+                    return True
+
+    def match_cat(cat_element):
```
also suggest add docstring

peixiu replied:
has updated, thanks~

peixiu commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,62 @@
-    signtool_cmd = "c:\Program Files\Microsoft SDKs\Windows\v7.0\Bin\signtool.exe" verify /v /kp /c %s %s
-    drive_list = F: A:
+    get_disk_letter_cmd = 'wmic logicaldisk | find "virtio-win"'
+    drive_list = A:
```
has updated, thanks~

peixiu commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,121 @@
+        if sub_os_name in os_name:
+            if arch_dir in plat_name:
+                if driver_dir in need_test_driver:
+                    return True
```
Try to change to your mentioned, like follows codes.  but no matter use () or \, the next line always need to 4 white space,  the Github CI check will report "E129 visually indented line with same indent as next logical line" error, so keep the currently code.
       if (sub_os_name in os_name and arch_dir in plat_name
            and driver_dir in need_test_driver):
            return True

kimi1978wy replied:
I have the same opinion with xiagao
```
if (sub_os_name in os_name and arch_dir in plat_name
   and driver_dir in need_test_driver):
     return True
```
If "and" in line2 is the same write space with "(" in line1 , there is no E129 error. You can try it if have other errors.

peixiu commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,133 @@
+    os_dirs = params.objects("os_dirs")
+    arch_dir = params.get("arch_dir")
     fails_log = """Failed signature check log:\n"""
+    results_all = """All the signature check log:\n"""
```
the results_all is used to save all sign check log, it doesn't only save one line string, and the end of program, it will include many lines, so used """.

peixiu commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,62 @@
-    signtool_install = "D:\autoit3.exe c:\winsdk.au3"
-    signtool_cmd = "c:\Program Files\Microsoft SDKs\Windows\v7.0\Bin\signtool.exe" verify /v /kp /c %s %s
-    drive_list = F: A:
+    get_disk_letter_cmd = 'wmic logicaldisk | find "virtio-win"'
```
updated, thanks~

lijinlijin commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,62 @@
+            os_dirs = Win10 w10
+        Win2016:
+            signtool = "WIN_UTILS:\signtool\signtool-HLK\Signtool_amd64_HLK.exe"
+            os_dirs = Win10 2k16
```
Maybe you can make signtool and guest name into a map,just as "guest_alias" in "single_driver_install.cfg" does.
So is the os_dirs.

peixiu replied:
lijin, yes, I also want to realize that, thanks~

peixiu commented:
Tested with latest code:
 (1/2) smp_2.8192m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Win8.x86_64.io-github-autotest-qemu.virtio_driver_sign_check: PASS (581.87 s)
 (2/2) smp_2.8192m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.virtio_driver_sign_check: PASS (190.67 s)

peixiu commented:
@lijinlijin @xiagao I updated the code,  Simplified the cfg configuration and add a def get_os_arch_path() in py to get the original os_dir and arch_dir value,  could you help to review it again? thanks a lot~

xiagao commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,32 @@
+            signtool = "WIN_UTILS:\signtool\signtool-HCK\Signtool_amd64_HCK.exe"
+        Win10, Win2016:
+            signtool = "WIN_UTILS:\signtool\signtool-HLK\Signtool_amd64_HLK.exe"
+    signtool_cmd = ${signtool} verify /v /kp /c %s %s
```
Is there some iproblem to move this line to line 15 ?

peixiu replied:
yes, the ${signtool} must be used behind signtool definition, if not, will cannot load the correct signtool string~

xiagao commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,32 @@
+    list_files_cmd = 'dir /s /b %s | find "%s"'
+    get_disk_letter_cmd = 'wmic logicaldisk | find "virtio-win"'
+    drive_volume = A:
+    tested_drivers = Balloon pvpanic viorng vioserial vioinput NetKVM netkvm vioscsi viostor   
```
I think miss "" for this parameter

peixiu replied:
xiagao, thanks, will update it~

peixiu commented:
Tested with the latest code:
 (1/1) smp_2.2048m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.virtio_driver_sign_check: PASS (431.93 s)

@lijinlijin @xiagao I've updated the code, streamlined the cfg configuration, could you help to review it again~
Thanks a lot~

peixiu commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,153 @@
+
+        if (os_name == os_path) and (plat_name == arch_path):
+            if driver_dir in need_test_driver:
+                return True
```
Hi hachen, 

this value will be used at cat_list = filter(filter_all_tested_cat_files, drivers['.cat']), if driver name match filter_all_tested_cat_files judge condition also means the def filter_all_tested_cat_files return True, it will be kept, if not, it will be filtered. that's why I return True on this def define.

peixiu commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,153 @@
+        os_arch_path_list = os_arch_path.split("\\")
+        return os_arch_path_list
+
+    def filter_all_tested_cat_files(cat_files):
```
In fact, all need test files will be saved to cat_list(in 119 lines), this def  define a rule that how to select  the correct cat files.

peixiu commented:
@xiagao @lijinlijin  could you help to review this patch again? thanks~

lijinlijin commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,26 @@
-    cdrom_virtio = isos/windows/virtio-win.iso
-    cdrom_sdk = isos/windows/GRMSDKX_EN_DVD.iso
+    cdroms += " virtio"
+    cdrom_virtio = isos/windows/virtio-win.latest_prewhql.iso
```
Why use virtio-win.latest_prewhql.iso here?
We also need to test this case with whql'ed versions

lijinlijin commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,26 @@
+    list_files_cmd = 'dir /s /b %s | find "%s"'
+    get_disk_letter_cmd = 'wmic logicaldisk | find "virtio-win"'
+    drive_volume = A:
+    tested_drivers = "Balloon pvpanic viorng vioserial vioinput NetKVM netkvm vioscsi viostor"
```
Two netkvm drivers' name?
NetKVM netkvm

peixiu replied:
lijin, yes, here need  both netkvm and NetKVM, it's due to the directory name is different on cdrom and floopy, define 2 names to make sure the correct directory can be searched when test no matter driver on cdrom or floopy. only the netkvm is special, so define them here.

lijinlijin commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,26 @@
+    guest_alias = "Win7-32:w7\x86,Win7-64:w7\amd64,Win8-32:w8\x86,Win8-64:w8\amd64,Win8-32.1:w8.1\x86,Win8-64.1:w8.1\amd64,Win10-32:w10\x86,Win10-64:w10\amd64,Win2008-sp2-32:2k8\x86,Win2008-sp2-64:2k8\amd64,Win2008-r2-64:2k8R2\amd64,Win2012-64:2k12\amd64,Win2012-64r2:2k12R2\amd64,Win2016-64:2k16\amd64"
+    guest_alias_fl = "Win7-32:i386\Win7,Win7-64:amd64\Win7,Win8-32:i386\Win8,Win8-64:amd64\Win8,Win8-32.1:i386\Win8.1,Win8-64.1:amd64\Win8.1,Win10-32:i386\Win10,Win10-64:amd64\Win10,Win2008-sp2-32:i386\Win2008,Win2008-sp2-64:amd64\Win2008,Win2008-r2-64:amd64\Win2008R2,Win2012-64:amd64\Win2012,Win2012-64r2:amd64\Win2012R2,Win2016-64:amd64\Win2016"
+    i386:
+        floppy_name = isos/windows/virtio-win.latest_prewhql.vfd.i386
```
same issues as above, do not limit it to prewhql version

lijinlijin commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,26 @@
+    i386:
+        floppy_name = isos/windows/virtio-win.latest_prewhql.vfd.i386
+    x86_64:
+        floppy_name = isos/windows/virtio-win.latest_prewhql.vfd.x86_64
```
same issues as above, do not limit it to prewhql version

peixiu commented:
Hi @lijinlijin Updated the code, and tested with new code, it can be passed, could you help to review it again, thanks a lot~

 (1/2) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.Win2008.x86_64.sp2.io-github-autotest-qemu.virtio_driver_sign_check: PASS (529.24 s)
 (2/2) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.virtio_driver_sign_check: PASS (348.97 s)

lijinlijin commented:
# git am 1128.patch
Applying: virtio drivers sign check Test
/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/.git/rebase-apply/patch:48: new blank line at EOF.
+
warning: 1 line adds whitespace errors.

lijinlijin commented:
And when I run windows 10 guest, signcheck will also check the floppy part,right?
the case will fail as no file will be found.
Could you add some flags so that windows 10 will not do floppy check?

lijinlijin commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,153 @@
+        error_context.context("Running SignTool check test in guest...",
+                              logging.info)
+        disk_result = session.cmd_output(get_disk_letter)
+        disk_letter = re.search(r'(\w:)\s', disk_result, re.M).group(1)
```
utils_misc.get_win_disk_vol() can get virtio-win disk letter.

peixiu replied:
try to optimize here, thanks~

lijinlijin commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,153 @@
                 except IndexError:
                     msg = "Fail to get Number of files successfully Verified"
-                    raise error.TestFail(msg)
+                    test.fail(msg)
```
Is this step necessary? Sounds impossible to meet this situation: signtool check pass but no suc_num returned

peixiu replied:
lijin, here code funcion is: when the result not include the "Number of files successfully Verified: ([0-9]*)", it will report msg = "Fail to get Number of files successfully Verified". The re_suc = "Number of files successfully Verified: ([0-9]*)" is a condition that judge the verify successfully.

lijinlijin replied:
In my understanding, if result does not include  "Number of files successfully Verified: ([0-9])", the error will be caught in line 126 and whole script will be interrupted, it will never execute these lines.

peixiu replied:
lijin, on 126 line, if status not 0, means the cmd cannot be executed completely, 
the 133 is under the cmd run completely, but the output not include the verify info, like some tool's error or print other none verify info, the status =0, but not means the verify successfully. this is an accurate screening condition.

lijinlijin replied:
Could you have a double confirm?
From my side, if verify fail(I give wrong file to the cmd), then the status!=0, it will be caught by line126:
Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-58.0-py2.7.egg/avocado_vt/test.py:407
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-58.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
    return fn(*args, **kwargs)
  File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/virtio_driver_sign_check.py", line 133, in run
    test.fail(msg)
  File "/usr/lib/python2.7/site-packages/avocado_framework-58.0-py2.7.egg/avocado/core/test.py", line 1029, in fail
    raise exceptions.TestFail(message)
TestFail: Fail command: D:\signtool\Signtool_amd64_HCK.exe verify /v /kp /c E:\Balloon/2k12R2/amd64/balloon.cat E:\Balloon/2k12R2/amd64/balloon.pdb. Output: 
Verifying: E:\Balloon/2k12R2/amd64/balloon.pdb

Number of files successfully Verified: 0
Number of warnings: 0
Number of errors: 1
SignTool Error: File not found in the specified catalog.
SignTool Error: File not valid: E:\Balloon/2k12R2/amd64/balloon.pdb

lijinlijin commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,153 @@
+            virtio_drivers = []
+
         if fail_num > 0:
             msg = "Following %s driver(s) signature checked failed." % fail_num
```
once a signtool check fail, the whole job will be interrupted, so fail_num value can be only be 0 or 1.
1.better to use if fail_num;
2.msg:   driver(s) ---> driver
3.line 143 to line 147 seems useless as line 126 to 128 has detected the error.

peixiu replied:
Hi lijin, 
Look at 125 line: "status, result = session.cmd_status_output(test_cmd)", 
we consider this situation:  the cmd completely(status return 0) and the the result include : Number of files successfully Verified: 0 (I hit this info before).
1. due to the status return 0, so the 126-128 line will not be execute; 
2. then we look 132 line, the suc_num will equal to 0, then we see 137 -139 lines, "if int(suc_num) != 1" equal true, the fail_num=1 due to execute the "fail_num += 1", 
3. next, the code will point to 122 line due to in a loop here, will execute to verify next file, when execute 132 line, also may hit the upper situation, if hit, the fail_num=2. it will verify the file one by one in the loop.
4. after all files be verified, 143-147 lines start be executed,  "if fail_num > 0:" will print "msg = "Following %s driver(s) signature checked failed." % fail_num......" if hit upper situation these codes will be executed.
5. The whole job will be interrupted  when hit these 2 situations:
1). when the cmd status !=0, this time mostly due to the tool file cannot be found or the cmd timeout, will trigger 128 line test.fail(msg).
2).  the cmd result don't have "Number of files successfully Verified: ([0-9]*)" this info, will trigger 135 line test.fail(msg).

lijinlijin replied:
Could you check my previous comment? If number of successfully verifed is 0, the status is not 0

peixiu replied:
Hi lijin, I tried similar test on my host, I understood wrong the status return value for def cmd_status_output, it's return 0 only the cmd run without any error, not execute completely.

For this patch update:
1. I need the file is verified one by one, so it cannot be interrupt by someone fail,  on 122 line, I cancel interrupt trigger, just show failed cmd and output info.  Totally failed result will use the last 143-147 lines code to trigger. It can be make sure all test can be finished.
2. Changed the disk_letter get way, thanks for you advice.
3. 133 -135 lines need to keep, it will re define the exception  IndexError of re.findall(). It's use to explain this error. 

Please help to review it again, thanks a lot~

PASS 1-smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.virtio_driver_sign_check

lijinlijin commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,153 @@
+                if status:
+                    msg = "Fail command: %s. Output: %s" % (test_cmd, result)
+                    error_context.context("Fail command: %s. Output: %s" %
+                                          (test_cmd, result), logging.info)
```
logging.error is more appropriate.

peixiu replied:
Thanks, will update~

lijinlijin commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,153 @@
                 except IndexError:
                     msg = "Fail to get Number of files successfully Verified"
-                    raise error.TestFail(msg)
+                    test.fail(msg)
```
If you don't expect job be interrupted by one file signcheck, I think you should not use test.fail() here.

lijinlijin commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,153 @@
+                    msg = "Fail command: %s. Output: %s" % (test_cmd, result)
+                    error_context.context("Fail command: %s. Output: %s" %
+                                          (test_cmd, result), logging.info)
                 results_all += result
```
better to add \n after add each result to results_all

peixiu replied:
lijin, with currently code, the result.log info like follows, there are 2 space lines between 2 results connection.
=================================================================
Successfully verified: A:\amd64\Win2008R2\netkvm.sys

Number of files successfully Verified: 1
Number of warnings: 0
Number of errors: 0


Verifying: A:\amd64\Win2008R2\netkvm.inf
File is signed in catalog: A:\amd64\Win2008R2\netkvm.cat
Hash of file (sha1): F6B3680CE9A1396DED7D5E09EBF2A8B2CBE5277A
=================================================================

peixiu commented:
and lijin, tried on win10-32, it can be passed, just run on the cdrom test, skip the floopy disk test. If the floopy for win10 modify to normally(same as others) on rhel8 phase, the code may no need to modify, if not, will modify the code to match it.

 (1/2) smp_2.4096m.repeat1.pre_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (2696.19 s)
 (2/2) smp_2.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.virtio_driver_sign_check: PASS (141.87 s)

peixiu commented:
@lijinlijin could you help to review it again? thanks a lot~

lijinlijin commented on qemu/tests/cfg/virtio_driver_sign_check.cfg:
```diff
@@ -1,16 +1,24 @@
+    i386:
+        floppy_name = isos/windows/virtio-win_x86.vfd.el7
+    x86_64:
+        floppy_name = isos/windows/virtio-win_amd64.vfd.el7
```
suggest use following floppy instead:
isos/windows/virtio-win_x86.vfd
isos/windows/virtio-win_amd64.vfd

lijinlijin commented on qemu/tests/virtio_driver_sign_check.py:
```diff
@@ -1,87 +1,152 @@
                 except IndexError:
                     msg = "Fail to get Number of files successfully Verified"
-                    raise error.TestFail(msg)
+                    logging.error(msg)
```
If this error happens, it's not caught in the end of this job.

peixiu replied:
has updated, thanks a lot~

peixiu commented:
Will re-write this function with another way, so close this patch.
